cupy/fallback_mode/fallback.py

```diff
@@ -0,0 +1,73 @@
+"""
+Main fallback class.
+"""
+import cupy as cp # NOQA
+import numpy as np # NOQA
+
+
+from cupy.fallback_mode.utils import FallbackUtil
+from cupy.fallback_mode.utils import get_last_and_rest
+from cupy.fallback_mode.utils import join_attrs
+from cupy.fallback_mode.utils import call_cupy
+from cupy.fallback_mode.utils import call_numpy
+
+
+class Recursive_attr:
+
+    def __init__(self, name):
+        self.name = name
+
+    def __getattr__(self, attr):
+        if self.name == 'numpy':
+            FallbackUtil.clear_attrs()
+        FallbackUtil.add_attrs(attr)
+        return dummy
+
+    def __call__(self, *args, **kwargs):
+        return fallback(*args, **kwargs)
+
+
+class Fallback(FallbackUtil):
+
+    def __call__(self, *args, **kwargs):
+        attributes = FallbackUtil.get_attr_list_copy()
+
+        sub_module, func_name = get_last_and_rest(attributes)
+
+        # trying cupy
+        try:
+            if sub_module == '':
+                cupy_path = 'cp'
+            else:
+                cupy_path = 'cp' + '.' + sub_module
```

Comment:
For the next version of code, you could abstract this into its own method since you use the same logic below.

Reply:
Sure I'll do that!